### PR TITLE
Add API controller base and route scope for APIs

### LINKED ISSUE #38: Setup API controller and token based authentication for API access by airflow
Airflow will be interacting with the application via an API that can update status, etc. We should follow our existing pattern of token based authentication for this access.

We have plenty of examples of this in our catalog, the recent DLME implementation is a good example:

```
module Api
  # A base for APIs that use simple token authentication
  class ApiController < ActionController::API
    include ActionController::HttpAuthentication::Token::ControllerMethods

    before_action :authenticate
    respond_to :json

    private

    def authenticate
      authenticate_or_request_with_http_token do |token, _options|
        ActiveSupport::SecurityUtils.secure_compare(token, ENV.fetch('API_TOKEN', ''))
      end
    end
  end
end
```

- [x] Add security scheme to `openapi.yml` (see also [sdr-api spec](https://github.com/sul-dlss/sdr-api/blob/main/openapi.yml#L2203-L2209), [OpenAPI authentication docs](https://swagger.io/docs/specification/authentication/))

### PR DESCRIPTION
**Pull request by mjgiarlo**
# Why was this change made? 🤔

Fixes #38

This commit adds a API controller class which is to be inherited by controllers providing APIs. The responsibilities of the base class are to handle token-based authentication (which is now documented in the application API spec) and to default to JSON responses.

# How was this change tested? 🤨

CI

### CODE REVIEW

File: app/controllers/api_controller.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+  private
+
+  def authenticate
+    authenticate_or_request_with_http_token do |token, _options|
+      # NOTE: We can inject the value of `API_TOKEN` via vault lookup in puppet
+      ActiveSupport::SecurityUtils.secure_compare(token, ENV.fetch("API_TOKEN", ""))
+    end
+  end
+end
```
Comment (mjgiarlo) [inline app/controllers/api_controller.rb:16]: Test coverage will improve when we add our first legit API controller.

File: config/routes.rb [hunk 2]
```diff
@@ -11,4 +11,8 @@
   resources :vendors, only: [:index, :show]
 
   root to: "stats#index"
+
+  scope "v1" do
+    # NOTE: API controller routes go in this block
+  end
```
Comment (ndushay) [inline config/routes.rb:17]: so our experience with prescat was that ... we can never change the version of the API because it's too cumbersome -- how do you have openapi for multiple versions?  How many consumers are there, really, and is it worth it?   And probably other reasons I don't remember from a few discussions with @jmartin-sul.  

How many consumers will this app have?  How will versioning be used?   It may not be worth it.

**Review by ndushay — COMMENTED** [review]
a comment for your consideration.

File: config/routes.rb (hunk 2 above)
Comment (mjgiarlo) [reply config/routes.rb:17]: @ndushay It does feel a bit YAGNI given we've never incremented an API version before within SDR. Currently, most of our API apps *do* use a scope to wrap API operations though, e.g.:

* DSA: `/v1`
* DIA: `/dor`
* prescat: `/v1`
* sdr-api: `/v1`
* techmd: `/v1`

I'm thinking that using `/v1` for consistency, weighed against what you wrote, is of limited value. But particular in an app that is part UI and part API, I do think it's helpful to reflect that certain URLs in the app are API-focused. (I probably couldn't defend that too strongly.)

Long story short: how about a scope of `/api` instead of `/v1`?

Comment (mjgiarlo) [reply config/routes.rb]: I made this change.

**Review by ndushay — APPROVED** [review]
